[SpecOps2016] iP

SpecOps2016 opened the pull request:
# DukePro
“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

-  text-based
-  easy to learn
-  ~~FAST~~ SUPER FAST to use
All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

SpecOps2016 committed:
Level-1

SpecOps2016 committed:
Level-2

SpecOps2016 committed:
Level-3

SpecOps2016 committed:
Level-4

SpecOps2016 committed:
Level-5

SpecOps2016 committed:
Level-6

SpecOps2016 committed:
Level-5 v2

SpecOps2016 committed:
Level-4, Level-5, Level-6 v2

SpecOps2016 committed:
Level A-TextUiTesting

SpecOps2016 committed:
Level A-TextUiTesting v2

SpecOps2016 committed:
Level-7

SpecOps2016 committed:
Level-8

SpecOps2016 committed:
Level-7 v2, Level-8 v2

SpecOps2016 committed:
A-MoreOOP, Level-7 and Level-8 v3

SpecOps2016 committed:
A-MorePackages

SpecOps2016 committed:
Merge branch 'add-gradle-support' of https://github.com/SpecOps2016/ip

# Conflicts:
#	build.gradle
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

SpecOps2016 committed:
A-Gradle

SpecOps2016 committed:
A-Junit

SpecOps2016 committed:
A-JavaDoc, A-CodingStandard

SpecOps2016 committed:
Level-9

SpecOps2016 committed:
Level-9 v2

SpecOps2016 committed:
A-JUnit v2

Evande1 commented on src/main/java/components/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+public class Deadline extends Task {
+
+  protected String by;
+  protected LocalDate by2;
```
Perhaps the variable name should be dateDueBy instead of by2?

chunkhaik replied:
Agreed! More clarity to variable names would help with your code readability!

Evande1 commented on src/main/java/components/Event.java:
```diff
@@ -0,0 +1,40 @@
+public class Event extends Task {
+
+  protected String at;
+  protected LocalDate at2;
```
Similarly, perhaps instead of at2, we could use dateOccuringAt?

Evande1 commented on src/main/java/components/Parser.java:
```diff
@@ -0,0 +1,194 @@
+      taskList.showTasks();
+    } else if (line.contains("unmark")) {
+      if (line.equals("unmark")) {
+        throw new DukeException("☹ OOPS!!! The description of a mark cannot be empty.");
```
Maybe we could edit the exception so we know where the error occurs? I.e. unmark instead of mark. Nice validation of inputs tho, keep it up! 😃

Evande1 commented on src/main/java/components/Storage.java:
```diff
@@ -0,0 +1,88 @@
+public class Storage {
+
+  private File f;
+  private ArrayList<Task> ls;
```
Perhaps we could change the name of ls to a pural form of collection of objects.

chunkhaik replied:
ditto comment, actualy nouns instead of shortforms would probably be useful too!

Evande1 reviewed (commented):
Generally really well done, easy on the eyes with good verb usage for function names and variables with sensible nouns.

bryanhce commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,36 @@
+import components.DukeException;
```
Should there be a package system implemented?

bryanhce commented on src/main/java/components/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+    this.by = by;
+  }
+
+  public Deadline(String description, LocalDate by2) {
```
Any reason you created 2 constructors instead of using just 1?

bryanhce commented on src/main/java/components/Parser.java:
```diff
@@ -0,0 +1,194 @@
+      } else {
+        try {
+          String description = line.substring(6, line.indexOf("/") - 1);
+          LocalDate d1 = LocalDate.parse(line.substring(line.indexOf("/") + 4));
```
Could the variable name d1 be more informative?

bryanhce commented on src/main/java/components/Parser.java:
```diff
@@ -0,0 +1,194 @@
+        } catch (DateTimeParseException e) {
+          String description = line.substring(6, line.indexOf("/") - 1);
+          String at = line.substring(line.indexOf("/") + 4);
+          Event test = new Event(description, at);
```
Could the variable test be more descriptive?

bryanhce commented on src/main/java/components/Task.java:
```diff
@@ -0,0 +1,49 @@
+  }
+
+  /**
+   * sets status of task to be completed or not.
```
Should the first letter of the JavaDocs be capitalised?

bryanhce commented on src/main/java/components/Task.java:
```diff
@@ -0,0 +1,49 @@
+   *
+   * @param value true represents completion while false represents incompletion.
+   */
+  public void setStatus(boolean value) {
```
Would it be better to have 2 separate methods to set the status of the task? Maybe by passing in the boolean value it exposes the implementation of the task class?

chunkhaik replied:
Agreed. Should probably have setTrue( ) and setFalse( ), it's better encapsulation as well

bryanhce commented on src/main/java/components/TaskList.java:
```diff
@@ -0,0 +1,108 @@
+  }
+
+  /**
+   * shows the list of tasks the user has inputted.
```
Should the JavaDocs description start with a capitalised word?

bryanhce reviewed (commented):
LGTM. Nice work! Everything seems to be working well, just some simple nits to fix. All the best! 😄

SpecOps2016 committed:
Fixed some bugs

SpecOps2016 committed:
Fixed some more bugs

chunkhaik commented on duke.txt:
```diff
@@ -0,0 +1,6 @@
+[T][ ] borrow book
+[D][ ] return book (by: Oct 15 2019)
+[T][ ] borror book
+[D][ ] return book (by: Feb 03 2019)
+[E][ ] do dishes (at: Snday)
+[T][ ] suck my dick
```
could probably add this file to .gitignore!

chunkhaik commented on src/main/java/components/Event.java:
```diff
@@ -0,0 +1,40 @@
+    if (this.at != null) {
+      return "[E]" + super.toString() + " (at: " + at + ")";
+    } else {
+      DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
+      return "[E]" + super.toString() + " (at: " + at2.format(formatter) + ")";
```
This formatting method could be better abstracted by the Parser class!

chunkhaik commented on src/main/java/components/Parser.java:
```diff
@@ -0,0 +1,195 @@
+        int num = Integer.parseInt(line.substring(7));
+        taskList.setTaskStatus(num - 1, false);
+      }
+    } else if (line.contains("mark")) {
```
This could potentially be a problem if the task given contains the word "mark" in it, similar to all the other case! Could probably check the first word instead of using contains

chunkhaik commented on src/main/java/components/Task.java:
```diff
@@ -0,0 +1,49 @@
+  }
+
+  /**
+   * get status icon of a task, with "X" representing completion and
+   * a blank representing incompletion.
```
style error with javadoc

chunkhaik reviewed (commented):
Overall great encapsulation! Your code was easy to understand too, just small mistakes here and there! The automated checkstyle would probably help!

But great work!

SpecOps2016 committed:
Level-10 + fixed bugs

SpecOps2016 committed:
A-Assertions

SpecOps2016 committed:
Implementing C-BetterSearch

The "find" command now gives suggestions for items that partially
matches the keywords rather than only giving suggestions for items
that exactly matches the keywords.

SpecOps2016 committed:
A-CodeQuality

SpecOps2016 committed:
Merge pull request #3 from SpecOps2016/branch-{CodeQuality}

Branch {Code Quality}

SpecOps2016 committed:
updated docs

SpecOps2016 committed:
fixed bugs in "find", "event"

SpecOps2016 committed:
fixed bugs in "find" and "event"

SpecOps2016 committed:
fixed assertions and other miscellaneous changes
